ASIM/dev/ASimTester/filteringTest/ASimFilteringTest.py

```diff
@@ -119,44 +120,74 @@ def get_substring_or_default(default, substring, rows, current_list):
     return default
 
 
-def get_prefix(str, rows, current_list):
+def get_prefix(str, rows, current_list, delimiter):
     '''
-    Returns the prefix of a string until its last dot, under certain conditions:
+    Returns the prefix of a string until its last occurrence of the delimiter, under certain conditions:
     - The prefix is not present in the 'current_list'.
     - The prefix is not contained in all of the values in rows.
-    If the string does not contain a dot or fails to meet the conditions above, the original string is returned.
+    If the string does not contain the delimiter or fails to meet the conditions above, the original string is returned.
     Example:
+        delimiter = '.'
         str = "example.com.subdomain"
         output = "example.com"
     '''
-    last_dot_index = str.rfind('.')
-    # If there is no dot in the string, return the string
-    if last_dot_index == -1:
+    last_delimiter_index = str.rfind(delimiter)
+    # If there is no delimiter in the string, return the string
+    if last_delimiter_index == -1:
         return str
     
-    substring = str[:last_dot_index]
+    substring = str[:last_delimiter_index]
     return get_substring_or_default(str, substring, rows, current_list)
 
 
-def get_postfix(str, rows, current_list):
+def get_postfix(str, rows, current_list, delimiter):
     '''
-    Returns the postfix of a string following its first dot, under certain conditions:
+    Returns the postfix of a string following the first occurrence of the delimiter, under certain conditions:
     - The postfix is not present in the 'current_list'.
     - The postfix is not contained in all of the values in rows.
-    If the string does not contain a dot or fails to meet the conditions above, the original string is returned.
+    If the string does not contain the delimiter or fails to meet the conditions above, the original string is returned.
     Example:
+        delimiter = '.'
         str = "example.com.subdomain"
         output = "com.subdomain"
     '''
-    first_dot_index = str.find('.')
-    # If there is no dot in the string, return the string
-    if first_dot_index == -1:
+    first_delimiter_index = str.find(delimiter)
+    # If there is no delimiter in the string, return the string
+    if first_delimiter_index == -1:
         return str
     
-    substring = str[first_dot_index + 1:]
+    substring = str[first_delimiter_index + 1:]
     return get_substring_or_default(str, substring, rows, current_list)
 
 
+def get_non_word_character_from_rows(rows):
+    '''
+    Returns the most frequent non-word character from the first 5 values in rows.
+    Example:
+        rows = ["123.456.4.1/4" , "44-55.5.1293", "11/33.99.13.12"]
+        output = '.' (as '.' appearers in the values of rows more than '/' and '-')
+    '''
+    character_count = {}
+    i = 0
+    number_of_rows_to_check = 5
+    # Counting each occurrence of a non-word character
+    for row in rows:
```

Comment:
>for row in rows: [](http://example.com/codeflow?start=4&length=16)

change to:
for i in range(min result)

or use:
for index, value in enumerate(my_list)

and use index instead of i